universal-application-tool-0.0.1/app/repository/ProgramRepository.java

```diff
@@ -27,16 +27,10 @@ public ProgramRepository(EbeanConfig ebeanConfig, DatabaseExecutionContext execu
     return supplyAsync(() -> ebeanServer.find(Program.class).findSet(), executionContext);
   }
 
-  public CompletionStage<Optional<Program>> lookupProgram(String name, long version) {
+  public CompletionStage<Optional<Program>> lookupProgram(String name) {
     return supplyAsync(
         () ->
-            Optional.ofNullable(
-                ebeanServer
-                    .find(Program.class)
-                    .where()
-                    .eq("name", name)
-                    .eq("version", version)
-                    .findOne()),
+            Optional.ofNullable(ebeanServer.find(Program.class).where().eq("name", name).findOne()),
```

Comment:
suggestion (non-blocking): may as well switch to querying by `id` while we're here, we're unlikely to have a reason to look up by `name`

Reply:
Done